internal/server/v1/variables/variables.go

```diff
@@ -0,0 +1,121 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package variables
+
+import (
+	"context"
+	"sort"
+
+	pb "github.com/datacommonsorg/mixer/internal/proto"
+	"github.com/datacommonsorg/mixer/internal/server/statvar"
+	"github.com/datacommonsorg/mixer/internal/util"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/datacommonsorg/mixer/internal/store"
+)
+
+// Variables implements API for Mixer.Variables.
+func Variables(
+	ctx context.Context,
+	in *pb.VariablesRequest,
+	store *store.Store,
+) (*pb.VariablesResponse, error) {
+	entity := in.GetEntity()
+	if entity == "" {
+		return nil, status.Error(codes.InvalidArgument, "Missing required arguments: entity")
```

Comment:
In theory this check is not necessary since the ESP will check that the path is not valid if "entity" is missing. This is more of URI error than argument error.

Reply:
Removed.